### ISSUE TITLE
Pug version deprecation

### ISSUE THREAD

**Issue by adammockor** [adammockor/pug-inheritance#11]
Hey @pure180,

do you remember how `deprecated` flag should work? (https://github.com/adammockor/pug-inheritance/blob/master/src/version.coffee#L17)

Because I am using pug 2.0.0-beta6 and that line sets `deprecated` to `true`. 
If `deprecated` is `true`, we are searching for `Include` instead of `RawInclude`, but pug uses `RawInclude`. So is the logic inverted here?

**Comment by pure180** [comment]
Yes, I'd do. I'll write an exact explanation later when I am at home..

**Comment by pure180** [comment]
To compare the PUG version string from `package.json`. we are [cleaning](https://github.com/adammockor/pug-inheritance/blob/master/src/version.coffee#L20) the string before the comparison and using the `semver.lt()` method ([semver](https://github.com/npm/node-semver)) to return the boolean, but this method don't considers the incremented version by the release type. 

So in this case the substring `-beta6` causes an error and sets the flag to `options.deprecated = true` ([https://github.com/adammockor/pug-inheritance/blob/master/src/index.coffee#L9](https://github.com/adammockor/pug-inheritance/blob/master/src/index.coffee#L9)).

We have some options:
1. We can implement the method `semver.inc(version, release)` to return the version incremented by the release type, or
2. we can set the flag `options.deprecated = true` as rewritable, to change this boolean manually as well, and
3. we can also remove the substring while cleaning the version string, by considering all existing  release types.

Guess you can check this by removing the substring `-beta6` in your `package.json`.

**Comment by adammockor** [comment]
We are essentially doing `semver.lt('1.10.0', '2.0.0-beta6')` which should be true and it is. But we don't want to set deprecated flag to true in this case.

So my point is that it should be compered with `semver.gt('1.10.0', '2.0.0-beta6')`.

**Comment by adammockor** [comment]
So, I managed to get difference between `RawInclude` and `Include`.

Basically  we uses in our codebase `include mixins (node type is 'RawInclude')`  instead of `include mixins.pug (node type is 'Include')`.

When this plugin has `deprecated` flag, we are creating inheritance tree checking only `Include`.

So I update our codebase to use `.pug` extensions everywhere and Pug wants to deprecated extensionless includes anyway based on this https://github.com/pugjs/pug/issues/2305#issuecomment-211321404 information.

But I am still confused about why we are not creating tree with `RawInclude`.

**Comment by pure180** [comment]
Sorry, I am a bit busy at the moment. I've to look into the pugwalk again, to answer your question. I'll have time that next weekend let's talk about that then?

**Comment by adammockor** [comment]
Yeah no problem at all.

**Comment by pure180** [comment]
Guess we should drop the deprecation support. #12
